app/models/mixins/relationship_mixin.rb

```diff
@@ -515,7 +515,8 @@ def fulltree_ids_arranged(*args)
 
   # Returns the records in the tree from the root arranged in a tree
   def fulltree_arranged(*args)
-    Relationship.arranged_rels_to_resources(fulltree_rels_arranged(*args))
+    class_specific_preloaders = args.last.delete(:class_specific_preloaders) if args.last.kind_of?(Hash)
```

Comment:
I was aware of this, but I was avoiding it since down the chain it is also called in `fulltree_rels_arranged`.  Instead of changing pulling out the `options` here, I was being explicit and only removing the single key I needed here.  Seemed less risky to not mutate args here when it was already being done later in `fulltree_rels_arranged`.

That said, I am not terribly partial to this line of thought here.  I guess the diff for doing this would then look like this:

```diff
   def fulltree_arranged(*args)
-    Relationship.arranged_rels_to_resources(fulltree_rels_arranged(*args))
+    options = args.extract_options!
+    class_specific_preloaders = options.delete(:class_specific_preloaders)
+    Relationship.arranged_rels_to_resources(fulltree_rels_arranged(*args, options), true, class_specific_preloaders)
   end
```

Does make that last line a bit longer, and I am all about shorter lines when possible, but should be the same.